app/models/product_purchase.rb

```diff
@@ -0,0 +1,19 @@
+class ProductPurchase < ApplicationRecord
+  self.abstract_class = true
+
+  belongs_to :product
+  belongs_to :child
+
+  before_validation :generate_user_facing_id
+  validates :shipping_name, :address, :zipcode, :product_id, :child_id, presence: true
+
+  def to_param
+    user_facing_id
+  end
+
+  private
+
+  def generate_user_facing_id
```

Comment:
Optionally auto-generate the user-facing id instead of requiring it be passed in on creation